### PR TITLE
Tooling rework - Added TS, Vite, Prettier + Eslint and refactored accordingly

### PR DESCRIPTION
**Pull request by jirijavora**
Enjoy! :v:

As the title says this PR is focused on updating tooling, but while I was rewriting everything it also seemed like a good idea to cleanup a few things, so there's been a some small refactor (+CSS update for better animations because I wanted it to look nice)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+export const APP_PADDING_X = 20;
+
+export const TIMETABLE_START_HOUR = 7;
+export const TIMETABLE_END_HOUR = 20;
+
+export const TIMETABLE_PIN_START_HOUR = 7.1;
+export const TIMETABLE_PIN_END_HOUR = 20.7;
+
+// FIXME: Rework this to not be so fufu...
```
Comment (jakcharvat) [inline src/App.tsx:18]: fufu

**Comment by jirijavora** [comment]
@sentisso Please remove the commit and put it into a separate branch (created from this one). Then create a new PR with the target branch being this one. (Separate functionality from a different contributor should have a separate PR in my opinion)

**Comment by sentisso** [comment]
@jirijavora tru

**Comment by jirijavora** [comment]
@sentisso I just force pushed to remove your changes from this PR. Please create the seperate PR as discussed.

**Comment by sentisso** [comment]
@jirijavora yes, sorry. I was busy shredding the hills. Thanks

**Pull request merged**